libs/langchain/langchain/retrievers/pinecone_hybrid_search.py

```diff
@@ -103,7 +210,7 @@ class PineconeHybridSearchRetriever(BaseRetriever):
     """description"""
     sparse_encoder: Any
     """Sparse encoder to use."""
-    index: Any
+    index_upsert: PineconeIndexUpsert
```

Comment:
Could you avoid making a breaking change here and instead do isinstance checks in the create index code?

Reply:
Sure, no problem. I'll make it work without any breaking changes.